ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java

```diff
@@ -71,6 +81,49 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  public @Nullable Typeface getTypeface(
+    String fontFamilyName,
+    int style,
+    Context context) {
+
+    if (!mTypeCacheLoaded) {
+      try {
+        ApplicationInfo app = context.getPackageManager().getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+        Bundle bundle = app.metaData;
+        int fontsId = bundle.getInt("rn_fonts", 0);
+        if (fontsId != 0) {
+          Resources resources = context.getResources();
+          TypedArray fonts = resources.obtainTypedArray(fontsId);
+          for (int i = 0; i < fonts.length(); i++) {
+            int fontId = fonts.getResourceId(i, 0);
+            if (fontId != 0) {
+              Typeface font = ResourcesCompat.getFont(context, fontId);
```

Comment:
How expensive is loading `Typeface` instances for every single font of the app into the cache and calling `resources.getResourceEntryName ` at startup?

If it is also taking too long (for FB scale), although it's verbose you might do something like:

```
<array name="fonts">
    
    <item>srisakdi</item>
    <item>@font/srisakdi</item>
    
    <item>anotherfont</item>
    <item>@font/anotherfont</item>

</array>
```

This way you can iterate the array in increments of 2, and get away with only caching `fontFamily`->`fontId` mapping at startup, and only inflate `fontId`->`Typeface` still lazily.

Reply:
Also this should allow font file names to not strictly match font family names, ex:

```
<item>Open Sans</item>
<item>@font/open_sans</item>
```

Native devs will be more familiar underscored naming `R.font.open_sans` whereas web developers will be more familiar with human readable naming `fontFamily: "Open Sans"`. 

This will allow sharing cross platform styling code with iOS, where font family name is already retrieved from font file content, not from file names.

Reply:
I tried to mimic iOS feature, where we only list font files, but also make typo-proof. Also I couldn't find anything about performance impact of getResourceEntryName, and hope it won't have impact on performance.

Reply:
iOS works a bit differently, it reads font family name from the file content, Android in your implementation reads from the file name.
https://developer.apple.com/documentation/uikit/text_display_and_fonts/adding_a_custom_font_to_your_app
https://stackoverflow.com/questions/16788330/how-do-i-get-the-font-name-from-an-otf-or-ttf-file

IMO this will force developers to write strange JS code like: 
`{fontFamily: Platform.OS == 'ios' ? 'Open Sans' : 'open_sans'}`
I dont know why typo-proof is such an important thing, you only need to match with what you use in JS code, you could make a mistake in font file name too.

---
Just looking at getResourceEntryName implementation in AOSP repo, it's implemented in C++ so you will at least jump between JNI bridge for each call.

Reply:
Yep, I found that Android doesn't allow to use font family name, but resource filename without extension. So difference is inevitable.

With this change we can use various font weights like medium easily. Currently, you have to write something like which makes text style sharing difficult.
```js
{
   fontFamily: Platform.OS == 'ios' ? 'Open Sans' : 'open_sans_medium'
   fontWeight: Platform.OS == 'ios' ? '500' : 'normal'
}
```

once PR lands you will write

```js
{
   fontFamily: Platform.OS == 'ios' ? 'Open Sans' : 'open_sans'
   fontWeight: '500'
}
```

Reply:
You can name your file "Open Sans.ttf", "Open Sans_bold.ttf"... etc though, so the code can already be cross platform `fontFamily: "Open Sans"`. Can you leave a space in these font resource file names and reuse that resource in Java? IMO it's not good practice to enforce android resource naming conventions into JS land.